Sources/HealthKitOnFHIR/HealthKit Extensions/HKCategorySample+Observation.swift

```diff
@@ -0,0 +1,84 @@
+//
+// This source file is part of the HealthKitOnFHIR open source project
+//
+// SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import HealthKit
+import ModelsR4
+
+
+extension HKCategorySample {
+    // Disabled the following rules as this function is readable
+    // swiftlint:disable:next cyclomatic_complexity function_body_length
+    func buildCategoryObservation(
+        _ observation: inout Observation,
+        mappings: HKSampleMapping = HKSampleMapping.default
+    ) throws {
+        guard let mapping = mappings.categorySampleMapping[self.categoryType] else {
+            throw HealthKitOnFHIRError.notSupported
+        }
+
+        let categoryTypeString = self.categoryType.description
+        
+        var valueString: String?
```

Comment:
The value is never mutated and can therefore be made a constant:
```suggestion
        let valueString: String?
```